### ISSUE TITLE
Basic Question about `get_batch_logps()`

### ISSUE THREAD

**Issue by vgoklani** [huggingface/trl#1172]
For the logits, why do we use `:-1` instead of `1:`

    labels = labels[:, 1:].clone()
    logits = logits[:, :-1, :]

as opposed to:

    logits = logits[:, 1:, :]

https://github.com/huggingface/trl/blob/main/trl/trainer/dpo_trainer.py#L861

**Comment by lvwerra** [comment]
The model predicts the _next_ token so labels(=inputs) and logits are shifted by one. Hope that clarifies it!

**Comment by younesbelkada** [comment]
Closing the issue! Feel free to re-open if you have more questions
